api/host.go

```diff
@@ -1,153 +1,227 @@
-func (srv *Server) hostDeleteHandler(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
-	writeJSON(w, "TODO: fix/revamp this call")
-	// hash, err := scanAddress(ps.ByName("filecontractid"))
-	// if err != nil {
-	// 	writeError(w, err.Error(), http.StatusBadRequest)
-	// 	return
-	// }
-	// fcid := types.FileContractID(hash)
-	// err = srv.host.DeleteContract(fcid)
-	// if err != nil {
-	// 	writeError(w, err.Error(), http.StatusBadRequest)
-	// 	return
-	// }
-	// writeSuccess(w)
+// storageHandler returns a bunch of information about storage management on
+// the host.
+func (srv *Server) storageHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	sfs, err := srv.host.StorageFolders()
+	if err != nil {
+		writeError(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	sg := StorageGET{
+		StorageFolderMetadata: sfs,
+	}
+	writeJSON(w, sg)
+}
+
+// storageFoldersAddHandler adds a storage folder to the storage manager.
+func (srv *Server) storageFoldersAddHandler(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
+	folderPath := ps.ByName("folder")
+	var folderSize uint64
+	_, err := fmt.Sscan(req.FormValue("size"), &folderSize)
+	if err != nil {
+		writeError(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	err = srv.host.AddStorageFolder(folderPath, folderSize)
+	if err != nil {
+		writeError(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	writeSuccess(w)
+}
+
+// storageFoldersResizeHandler resizes a storage folder in the storage manager.
+func (srv *Server) storageFoldersResizeHandler(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
+	folderPath := ps.ByName("folder")
+	storageFolders, err := srv.host.StorageFolders()
+	if err != nil {
+		writeError(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	folderIndex, err := folderIndex(folderPath, storageFolders)
+	if err != nil {
+		writeError(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	var newSize uint64
+	_, err = fmt.Sscan(req.FormValue("newsize"), &newSize)
+	if err != nil {
+		writeError(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	err = srv.host.ResizeStorageFolder(folderIndex, newSize)
+	if err != nil {
+		writeError(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	writeSuccess(w)
+}
+
+// storageFoldersRemoveHandler removes a storage folder from the storage
+// manager.
+func (srv *Server) storageFoldersRemoveHandler(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
+	folderPath := ps.ByName("folder")
+	storageFolders, err := srv.host.StorageFolders()
+	if err != nil {
+		writeError(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	folderIndex, err := folderIndex(folderPath, storageFolders)
+	if err != nil {
+		writeError(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	var force bool
+	_, err = fmt.Sscan(req.FormValue("force"), &force)
+	if err != nil {
+		writeError(w, err.Error(), http.StatusBadRequest)
```

Comment:
this means `force` is required in all requests. Better would be:
```go
force := false
if req.FormValue("force") != "" {
    _, err = fmt.Sscan(req.FormValue("force"), &force)
}
```
or even simply:
```go
force := req.FormValue("force") == "true"
```
which is the approach used by the renter upload API call.